### PR TITLE
Support docker 

### LINKED ISSUE #13: Support Docker
Apply to add docker support so that interested users can quickly start and test locally. A simple Dockerfile implementation is provided. If there are other requirements for the image, such as size restrictions and features, I will accept it.

### PR DESCRIPTION
**Pull request by EurusEurus**
Add Dockerfile, use gosu run specific application as specific user.

**What problem does this PR solve?**
Issue Number: close #13 

**What is changed and how it works?**
- Dockerfile 
- docker-entrypoint.sh 
- .dockerignore

### CODE REVIEW

**Comment by queensferryme** [comment]
Thanks for your contribution! Here are two of my concerns:

- The `.dockerignore` seems incomplete (compared to `.gitignore`). For example the document build artifact `site/` is not excluded from the docker image.
- I don't quite understand `gosu` and `docker-entrypoint.sh`, could you provide some context or explanation?

**Comment by EurusEurus** [comment]
Q1 `.dockerignore`
I didn't read the code completely, and the files that might be generated by the script were not removed.

Q2 `gosu`
Avoid using the root user to start the application, and can still handle users and groups after the image is built. `gosu` can intercept the command matched in cmd and execute it with the specified user instead.

ref.
repo https://github.com/tianon/gosu 
demo repo https://github.com/sudo-bmitch/jenkins-docker

> Thanks for your contribution! Here are two of my concerns:
> 
> * The `.dockerignore` seems incomplete (compared to `.gitignore`). For example the document build artifact `site/` is not excluded from the docker image.
> * I don't quite understand `gosu` and `docker-entrypoint.sh`, could you provide some context or explanation?

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+README.md
+README.zh.md
+Dockerfile
+mypy.ini
+
+# whl
+dist/
+
+# Docs
+site/
```
Comment (EurusEurus) [inline .dockerignore:13]: update `.dockerignore`,  add the content in .gitignore.

PTAL @queensferryme

**Comment by queensferryme** [comment]
I made quite some changes to the PR:

- use alpine python image, which reduce image size greatly;
  ```
  REPOSITORY            TAG          IMAGE ID       CREATED          SIZE
  rsserpent             latest       fea89a26b202   12 minutes ago   235MB
  rsserpent             0.1.3        eba50e4d545f   6 hours ago      1.07GB
  ```
- use `poetry install --no-dev`, also reducing image size;
- only copy `pyproject.toml`, `poetry.lock`, and `rsserpent/`, which helps keeping `.dockerignore` clean;
- use `COPY --chown` syntax, removing `docker-entrypoint.sh`.

I am not so sure about the removal of `docker-entrypoint.sh`. In my understanding, your `docker-entrypoint.sh` uses `gosu` to change the permissions of workdir. To me `COPY --chown` seems like a better solution, since we don't need to introduce `gosu` and `docker-entrypoint.sh`.

I will merge if @EurusEurus are okay with these changes.

**Comment by EurusEurus** [comment]
> I made quite some changes to the PR:
> 
> * use alpine python image, which reduce image size greatly;
>   ```
>   REPOSITORY            TAG          IMAGE ID       CREATED          SIZE
>   rsserpent             latest       fea89a26b202   12 minutes ago   235MB
>   rsserpent             0.1.3        eba50e4d545f   6 hours ago      1.07GB
>   ```
> * use `poetry install --no-dev`, also reducing image size;
> * only copy `pyproject.toml`, `poetry.lock`, and `rsserpent/`, which helps keeping `.dockerignore` clean;
> * use `COPY --chown` syntax, removing `docker-entrypoint.sh`.
> 
> I am not so sure about the removal of `docker-entrypoint.sh`. In my understanding, your `docker-entrypoint.sh` uses `gosu` to change the permissions of workdir. To me `COPY --chown` seems like a better solution, since we don't need to introduce `gosu` and `docker-entrypoint.sh`.
> 
> I will merge if @EurusEurus are okay with these changes.

LGTM @queensferryme 
ps.
```
$ docker-slim build rsserpent:5.3
rsserpent.slim           latest    1c0fe25629b7   18 seconds ago   115MB
```
https://dockersl.im/

**Comment by queensferryme** [comment]
Thanks. I will use docker-slim in github actions.

**Pull request merged**